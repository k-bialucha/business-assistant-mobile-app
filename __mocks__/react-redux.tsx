import React from 'react';
import { Text, View } from 'react-native';

export const Provider: React.FC<{}> = ({ children }) => (
  <View>
    <Text>Provider</Text>
    <View>{children}</View>
  </View>
);

export const useDispatch = jest.fn().mockImplementation(() => jest.fn());

export const useSelector = jest.fn().mockImplementation(() => ({}));
