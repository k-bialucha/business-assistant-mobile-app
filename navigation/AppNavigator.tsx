import React from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';

import DashboardScreen from '../screens/DashboardScreen';
import SalesScreen from '../screens/SalesScreen';
import CostsScreen from '../screens/CostsScreen';
import CostsEntryScreen from '../screens/CostsEntryScreen';
import SalesEntryScreen from '../screens/SalesEntryScreen';

const DashboardStackNavigator = createStackNavigator();

const DashboardNavigator = () => {
  return (
    <DashboardStackNavigator.Navigator>
      <DashboardStackNavigator.Screen
        name="Dashboard"
        component={DashboardScreen}
      />
    </DashboardStackNavigator.Navigator>
  );
};

const SalesStackNavigator = createStackNavigator();

const SalesNavigator = () => {
  return (
    <SalesStackNavigator.Navigator>
      <SalesStackNavigator.Screen name="SalesList" component={SalesScreen} />
      <SalesStackNavigator.Screen
        name="SalesEntry"
        component={SalesEntryScreen}
      />
    </SalesStackNavigator.Navigator>
  );
};

const CostsStackNavigator = createStackNavigator();

const CostsNavigator = () => {
  return (
    <CostsStackNavigator.Navigator>
      <CostsStackNavigator.Screen name="CostsList" component={CostsScreen} />
      <CostsStackNavigator.Screen
        name="CostsEntry"
        component={CostsEntryScreen}
      />
    </CostsStackNavigator.Navigator>
  );
};

const AppTabNavigator = createBottomTabNavigator();

const AppNavigator = () => {
  return (
    <AppTabNavigator.Navigator>
      <AppTabNavigator.Screen name="Dashboard" component={DashboardNavigator} />
      <AppTabNavigator.Screen name="Sales" component={SalesNavigator} />
      <AppTabNavigator.Screen name="Costs" component={CostsNavigator} />
    </AppTabNavigator.Navigator>
  );
};

export default AppNavigator;
