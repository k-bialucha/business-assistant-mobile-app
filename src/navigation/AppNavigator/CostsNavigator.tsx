import React from 'react';

import { CompositeNavigationProp, RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import CostsEntryScreen from '~/screens/CostsEntryScreen';
import CostsListScreen, {
  CostsListScreenNavOptions,
} from '~/screens/CostsListScreen';

import defaultScreenOptions from './defaultScreenOptions';
import {
  AppNavigatorNavigationProp,
  NavigationData as AppNavigatorNavigationData,
} from './index';

type ParamList = {
  CostsList: undefined;
  CostsEntry: undefined;
};

const CostsStackNavigator = createStackNavigator<ParamList>();

type Props = AppNavigatorNavigationData<'Costs'>;

const CostsNavigator: React.FC<Props> = () => {
  return (
    <CostsStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <CostsStackNavigator.Screen
        name="CostsList"
        component={CostsListScreen}
        options={CostsListScreenNavOptions}
      />
      <CostsStackNavigator.Screen
        name="CostsEntry"
        component={CostsEntryScreen}
      />
    </CostsStackNavigator.Navigator>
  );
};

export default CostsNavigator;

// combine navigation prop for nested navigators
// (AppNavigator data + this StackNavigator data)
export type NavigationPropCombined<
  RouteName extends keyof ParamList
> = CompositeNavigationProp<
  AppNavigatorNavigationProp<'Costs'>,
  StackNavigationProp<ParamList, RouteName>
>;

// export type so screens can get typing
export type NavigationData<RouteName extends keyof ParamList> = {
  navigation: NavigationPropCombined<RouteName>;
  route: RouteProp<ParamList, RouteName>;
};
