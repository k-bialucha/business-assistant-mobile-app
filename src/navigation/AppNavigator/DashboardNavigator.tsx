import React from 'react';

import { CompositeNavigationProp, RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import DashboardScreen from '~/screens/DashboardScreen';

import defaultScreenOptions from './defaultScreenOptions';
import {
  AppNavigatorNavigationProp,
  NavigationData as AppNavigatorNavigationData,
} from './index';

type ParamList = {
  Dashboard: undefined;
};

const DashboardStackNavigator = createStackNavigator<ParamList>();

type Props = AppNavigatorNavigationData<'Dashboard'>;

const DashboardNavigator: React.FC<Props> = () => {
  return (
    <DashboardStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <DashboardStackNavigator.Screen
        name="Dashboard"
        component={DashboardScreen}
      />
    </DashboardStackNavigator.Navigator>
  );
};

export default DashboardNavigator;

// combine navigation prop for nested navigators
// (AppNavigator data + this StackNavigator data)
export type NavigationPropCombined<
  RouteName extends keyof ParamList
> = CompositeNavigationProp<
  AppNavigatorNavigationProp<'Dashboard'>,
  StackNavigationProp<ParamList, RouteName>
>;

// export type so screens can get typing
export type NavigationData<RouteName extends keyof ParamList> = {
  navigation: NavigationPropCombined<RouteName>;
  route: RouteProp<ParamList, RouteName>;
};
