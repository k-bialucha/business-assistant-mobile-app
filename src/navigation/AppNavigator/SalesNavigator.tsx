import React from 'react';

import { CompositeNavigationProp, RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import SalesEntryScreen from '~/screens/SalesEntryScreen';
import SalesScreen, { SalesScreenNavOptions } from '~/screens/SalesScreen';

import defaultScreenOptions from './defaultScreenOptions';
import {
  AppNavigatorNavigationProp,
  NavigationData as AppNavigatorNavigationData,
} from './index';

type ParamList = {
  SalesList: undefined;
  SalesEntry: undefined;
};

const SalesStackNavigator = createStackNavigator<ParamList>();

type Props = AppNavigatorNavigationData<'Sales'>;

const SalesNavigator: React.FC<Props> = () => {
  return (
    <SalesStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <SalesStackNavigator.Screen
        name="SalesList"
        component={SalesScreen}
        options={SalesScreenNavOptions}
      />
      <SalesStackNavigator.Screen
        name="SalesEntry"
        component={SalesEntryScreen}
      />
    </SalesStackNavigator.Navigator>
  );
};

export default SalesNavigator;

// combine navigation prop for nested navigators
// (AppNavigator data + this StackNavigator data)
export type NavigationPropCombined<
  RouteName extends keyof ParamList
> = CompositeNavigationProp<
  AppNavigatorNavigationProp<'Sales'>,
  StackNavigationProp<ParamList, RouteName>
>;

// export type so screens can get typing
export type NavigationData<RouteName extends keyof ParamList> = {
  navigation: NavigationPropCombined<RouteName>;
  route: RouteProp<ParamList, RouteName>;
};
