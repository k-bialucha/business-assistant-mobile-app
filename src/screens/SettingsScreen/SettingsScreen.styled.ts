import styled from 'styled-components/native';

const StyledView = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`;

export { StyledView };
