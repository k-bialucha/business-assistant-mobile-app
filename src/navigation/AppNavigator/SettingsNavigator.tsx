import React from 'react';

import { CompositeNavigationProp, RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import SettingsScreen from '~/screens/SettingsScreen';
import UserSettingsScreen from '~/screens/UserSettingsScreen';
import { UserSettingsScreenNavOptions } from '~/screens/UserSettingsScreen/UserSettingsScreen';

import defaultScreenOptions from './defaultScreenOptions';
import {
  AppNavigatorNavigationProp,
  NavigationData as AppNavigatorNavigationData,
} from './index';

type ParamList = {
  Settings: undefined;
  UserSettings: undefined;
};

const SettingsStackNavigator = createStackNavigator<ParamList>();

type Props = AppNavigatorNavigationData<'Settings'>;

const SettingsNavigator: React.FC<Props> = () => {
  return (
    <SettingsStackNavigator.Navigator screenOptions={defaultScreenOptions}>
      <SettingsStackNavigator.Screen
        name="Settings"
        component={SettingsScreen}
      />
      <SettingsStackNavigator.Screen
        name="UserSettings"
        component={UserSettingsScreen}
        options={UserSettingsScreenNavOptions}
      />
    </SettingsStackNavigator.Navigator>
  );
};

export default SettingsNavigator;

// combine navigation prop for nested navigators
// (AppNavigator data + this StackNavigator data)
export type NavigationPropCombined<
  RouteName extends keyof ParamList
> = CompositeNavigationProp<
  AppNavigatorNavigationProp<'Settings'>,
  StackNavigationProp<ParamList, RouteName>
>;

// export type so screens can get typing
export type NavigationData<RouteName extends keyof ParamList> = {
  navigation: NavigationPropCombined<RouteName>;
  route: RouteProp<ParamList, RouteName>;
};
