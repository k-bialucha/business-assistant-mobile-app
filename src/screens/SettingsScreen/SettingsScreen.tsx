import React from 'react';
import { Alert, TouchableWithoutFeedback } from 'react-native';

import { ListItem } from 'react-native-elements';
import { useDispatch, useSelector } from 'react-redux';

import { LanguageSwitcher } from '~/components/LanguageSwitcher';
import { useAppTranslation } from '~/hooks/useAppTranslation';
import { NavigationData } from '~/navigation/AppNavigator/SettingsNavigator';
import { logout } from '~/redux/auth';
import { RootState } from '~/redux/rootReducer';
import Colors from '~/theme/Colors';

import { StyledView } from './SettingsScreen.styled';

type Props = NavigationData<'Settings'>;

const SettingsScreen: React.FC<Props> = ({ navigation }) => {
  const dispatch = useDispatch();
  const { username, userImage } = useSelector((state: RootState) => state.auth);
  const { t } = useAppTranslation();

  if (!username) return null;

  return (
    <StyledView>
      <TouchableWithoutFeedback
        onPress={() => navigation.navigate('UserSettings')}
      >
        <ListItem
          leftAvatar={{
            rounded: true,
            ...(userImage ? { source: { uri: userImage } } : {}),
          }}
          title={username}
          titleStyle={{
            color: Colors.navyBlue,
            fontSize: 20,
          }}
          style={{ marginTop: 35, marginBottom: 35 }}
          chevron
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={() => Alert.alert('"Open view screen with some settings"')}
      >
        <ListItem
          title="Test Option 1"
          titleStyle={{
            color: Colors.navyBlue,
          }}
          chevron
          style={{ marginBottom: 5 }}
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={() => Alert.alert('"Open view screen with some settings"')}
      >
        <ListItem
          title="Test Option 2"
          titleStyle={{
            color: Colors.navyBlue,
          }}
          style={{ marginBottom: 5 }}
          chevron
        />
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={() => {
          dispatch(logout());
        }}
      >
        <ListItem
          title={t('Logout')}
          titleStyle={{
            color: Colors.errorRed,
          }}
          style={{ marginTop: 30 }}
          contentContainerStyle={{ alignItems: 'center' }}
        />
      </TouchableWithoutFeedback>
      <LanguageSwitcher />
    </StyledView>
  );
};

export default SettingsScreen;
