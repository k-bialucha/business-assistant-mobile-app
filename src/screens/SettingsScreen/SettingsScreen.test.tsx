import React from 'react';

import { shallow } from 'enzyme';
import { useSelector } from 'react-redux';

import { NavigationData } from '~/navigation/AppNavigator/SettingsNavigator';

import SettingsScreen from './SettingsScreen';

(useSelector as jest.Mock).mockReturnValue({
  username: 'dummy-username',
  userImage: 'dummy-image-url',
});

type Props = NavigationData<'Settings'>;

const fakeProps: Props = {
  route: { key: '1234', name: 'Settings' },
  // @ts-ignore
  navigation: {},
};

describe('<SettingsScreen />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<SettingsScreen {...fakeProps} />);

    expect(tree).toMatchSnapshot();
  });
});
