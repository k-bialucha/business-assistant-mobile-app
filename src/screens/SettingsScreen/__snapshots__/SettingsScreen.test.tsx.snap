// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> matches the snapshot 1`] = `
<Styled(View)>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Themed.ListItem
      chevron={true}
      leftAvatar={
        Object {
          "rounded": true,
          "source": Object {
            "uri": "dummy-image-url",
          },
        }
      }
      style={
        Object {
          "marginBottom": 35,
          "marginTop": 35,
        }
      }
      title="dummy-username"
      titleStyle={
        Object {
          "color": "#1A2A40",
          "fontSize": 20,
        }
      }
    />
  </TouchableWithoutFeedback>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Themed.ListItem
      chevron={true}
      style={
        Object {
          "marginBottom": 5,
        }
      }
      title="Test Option 1"
      titleStyle={
        Object {
          "color": "#1A2A40",
        }
      }
    />
  </TouchableWithoutFeedback>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Themed.ListItem
      chevron={true}
      style={
        Object {
          "marginBottom": 5,
        }
      }
      title="Test Option 2"
      titleStyle={
        Object {
          "color": "#1A2A40",
        }
      }
    />
  </TouchableWithoutFeedback>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Themed.ListItem
      contentContainerStyle={
        Object {
          "alignItems": "center",
        }
      }
      style={
        Object {
          "marginTop": 30,
        }
      }
      title="[Translation: <Logout>]"
      titleStyle={
        Object {
          "color": "#ff0000",
        }
      }
    />
  </TouchableWithoutFeedback>
  <LanguageSwitcher />
</Styled(View)>
`;
